python/ray/data/datasource/datasink.py

```diff
@@ -26,20 +29,16 @@ def write(
         self,
         blocks: Iterable[Block],
         ctx: TaskContext,
-    ) -> Any:
+    ) -> None:
         """Write blocks. This is used by a single write task.
 
         Args:
             blocks: Generator of data blocks.
             ctx: ``TaskContext`` for the write task.
-
-        Returns:
-            A user-defined output. Can be anything, and the returned value is passed to
-            :meth:`~ray.data.Datasink.on_write_complete`.
         """
         raise NotImplementedError
 
-    def on_write_complete(self, write_results: List[Any]) -> None:
+    def on_write_complete(self, raw_write_results: List[Block]) -> None:
         """Callback for when a write job completes.
 
         This can be used to "commit" a write output. This method must
```

Comment:
Docstring description for `write_results` is out of date (`write` returns None now)
> The objects returned by every
                :meth:`~ray.data.Datasink.write` task.